python/astro_metadata_translator/translators/sdss.py

```diff
@@ -0,0 +1,205 @@
+# This file is part of astro_metadata_translator.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the LICENSE file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# Use of this source code is governed by a 3-clause BSD-style
+# license that can be found in the LICENSE file.
+
+"""Metadata translation code for SDSS FITS headers"""
+
+__all__ = ("SdssTranslator", )
+
+import posixpath
+
+from astropy.coordinates import EarthLocation, Angle, AltAz
+import astropy.units as u
+
+from ..translator import cache_translation, CORRECTIONS_RESOURCE_ROOT
+from .fits import FitsTranslator
+from .helpers import altaz_from_degree_headers,\
+    tracking_from_degree_headers
+
+
+class SdssTranslator(FitsTranslator):
+    """Metadata translator for SDSS standard headers.
+    NB: calibration data is not handled as calibration frames were
+    not available to me at time of writing.
+    """
+
+    name = "SDSS"
+    """Name of this translation class"""
+
+    supported_instrument = "Imager"
+    """Supports the SDSS imager instrument."""
+
+    default_resource_root = posixpath.join(CORRECTIONS_RESOURCE_ROOT, "SDSS")
+    """Default resource path root to use to locate header correction files."""
+
+    # SDSS has has a rotator, but in drift scan mode, the instrument
+    # angle on sky is set to +X=East, +Y=North which we define as a
+    # 0 degree rotation.
+    _const_map = {"boresight_rotation_angle": Angle(0*u.deg),
+                  "boresight_rotation_coord": "sky",
+                  "dark_time": 0.0*u.s,  # Drift scan impies no dark time
+                  "instrument": "Imager",  # We only ever ingest data from the imager
+                  "relative_humidity": None,
+                  "temperature": None,
+                  "pressure": None,
+                  "detector_serial": "UNKNOWN",
+                  }
+
+    _trivial_map = {"exposure_time": ("EXPTIME", dict(unit=u.s)),
+                    "object": "OBJECT",
+                    "physical_filter": "FILTER",
+                    "exposure_id": "DATASUM",
+                    "visit_id": "DATASUM",
+                    "science_program": "OBJECT",  # This is the closest I can think of to a useful program
+                    "detector_name": "CCDLOC",  # This is a numeric incoding of the "slot", i.e. filter+camcol
+                    "telescope": ("TELESCOP", dict(default="2.5m")),
```

Comment:
I'm a little concerned that telescope=2.5m and instrument="imager" is not very helpful for someone who solely has an ObservationInfo translated form of the header. Wikipedia calls the telescope "SDSS 2.5m" so we either call it that or call it "APO 2.5m". "Imager on SDSS 2.5m" seems more descriptive.

Reply:
OK.  I'll put both in the const map rather than trying to discern it from the header.

Reply:
It might be worth including TELESCOP in your `can_translate` testing though.

Reply:
You mean test for "2.5m"?

Reply:
Yes. Not important though. If SDSS only comes from 2.5m and ORIGIN has SDSS then I'm sure it's fine.

Reply:
I put it in anyway.